internal/aws/k8s/k8sclient/obj_store.go

```diff
@@ -0,0 +1,153 @@
+// Copyright  OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package k8sclient
+
+import (
+	"log"
+	"sync"
+
+	"k8s.io/apimachinery/pkg/api/meta"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+type ObjStore struct {
+	sync.RWMutex
+
+	refreshed bool
+	objs      map[types.UID]interface{}
+
+	transformFunc func(interface{}) (interface{}, error)
+}
+
+func NewObjStore(transformFunc func(interface{}) (interface{}, error)) *ObjStore {
+	return &ObjStore{
+		transformFunc: transformFunc,
+		objs:          map[types.UID]interface{}{},
+	}
+}
+
+// Track whether the underlying data store is refreshed or not.
+// Calling this func itself will reset the state to false.
```

Comment:
Having a Getter type of method name resetting state is fragile, can it be improved?